typespec-tests/src/main/java/com/cadl/naming/models/BytesData.java

```diff
@@ -17,6 +17,12 @@
  */
 @Immutable
 public final class BytesData extends Data {
+    /*
+     * The @data.kind property.
+     */
+    @Generated
+    private String type = "bytes";
```

Comment:
My origin attempt is to avoid this `String type` class variable, and directly provide
```java
    @Generated
    @Override
    public String getType() {
        return "bytes";
    }
```
(there is a corner case on behavior difference between the 2, on a case of user directly do `BinaryData("<invalid-bytes-data-json>").toObject(BytesData.class)` with a JSON of unknown type)

But reverted to the shadow variable.
Reason is `fromJson` have code on either `deserializedBytesData.setType(..)` or `deserializedBytesData.type = ..`, and I don't want to disrupt that part of code.